services/dqa-secondary-reports/main.py

```diff
@@ -0,0 +1,271 @@
+import codecs
+from datetime import datetime
+import git
+import json
+import logging
+import os
+import shutil
+import sys
+import uuid
+
+from requests.exceptions import HTTPError
+from flask import Flask, Response
+from flask.ext.cors import CORS
+
+from dictionary_parser import Parser as DictionaryParser
+from report_parser import Parser, logger
+from timer import ResourceTimer
+
+# Token for GitHub authorization. Set on startup.
+GITHUB_AUTH_TOKEN = None
+
+DIR_NAME = 'dqa_repo'
+REMOTE_URL = 'github.com/PEDSnet/Data-Quality.git'
+
+
+class DQAResource():
+    def __init__(self):
+        self.field_totals = {}
+        self.table_totals = {}
+        self.site_totals = {}
+        self.dictionary = {}
+
+        self.dir_name = DIR_NAME
+        self.remote_url = 'https://' + GITHUB_AUTH_TOKEN + '@' + REMOTE_URL
+
+    def process_dqa(self):
+        self.field_totals = {}
+        self.table_totals = {}
+        self.site_totals = {}
+
+        dir_name = './dqa_repo/SecondaryReports/'
+        hospitals = filter(
+            lambda f: (f != 'Ranking'
+                       and os.path.isdir(os.path.join(dir_name, f))),
+            os.listdir(dir_name))
+
+        self.site_totals = {}
+
+        for hospital in hospitals:
+            hosp_path = os.path.join(dir_name, hospital)
+
+            # for each site, we only need data from the latest report for each
+            # data model version, so go through the dqa reports in reverse
+            # order and ignore older reports for each data model version.
+            for etl_version in sorted(os.listdir(hosp_path), reverse=True):
+                etl_path = os.path.join(hosp_path, etl_version)
+
+                if etl_version.startswith('ETL') and os.path.isdir(etl_path):
+                    file_names = list(filter(lambda x: x.endswith('.csv'),
```

Comment:
> I had people tell me that using loops instead of filters means you haven't done any programming in the 21st century, so I try to squeeze them in when I remember :)

I hope *those people* are using lazily evaluated iterators otherwise they are wasting a lot of CPU cycles.